src/Avalonia.Controls/ToolTip.cs

```diff
@@ -106,24 +106,28 @@ private static void ShowToolTip(Control control)
             if (control != null && control.IsVisible && control.GetVisualRoot() != null)
             {
                 var cp = (control.GetVisualRoot() as IInputRoot)?.MouseDevice?.GetPosition(control);
-                var position = control.PointToScreen(cp ?? new Point(0, 0)) + new Vector(0, 22);
 
-                if (s_popup == null)
+                if (cp.HasValue && control.IsVisible && new Rect(control.Bounds.Size).Contains(cp.Value))
```

Comment:
If mouse is captured by another control, we shouldn't show the tooltip. So checking for IsPointerOver is correct behavior, I think